src/features/categories/category-selection/CategorySelectionModal.tsx

```diff
@@ -1,121 +1,125 @@
+import { ArrowLeft } from '@phosphor-icons/react';
+import { IndexableType } from 'dexie';
+import { useContext, useMemo, useState } from 'react';
+import { CategoriesContext, ICategoriesContext } from '../CategoriesContext';
+import CategoryModal from '../category-edit-modal/CategoryModal';
+import CategoryBadge from './category-badge/CategoryBadge';
+import CategoryItem from './CategoryItem';
+import { CategorySelectionFunction } from './CategorySelect';
+import classes from './CategorySelect.module.scss';
 
 type CategorySelectionModalProps = {
-  onClose: () => void
-  categoryIds: IndexableType[]
-  onCategoriesChange: (categories: IndexableType[]) => void
-}
+  onClose: () => void;
+  categoryIds: IndexableType[];
+  onCategoriesChange: (categories: IndexableType[]) => void;
+};
 
 const CategorySelectionModal = ({
   onClose,
   categoryIds,
   onCategoriesChange,
 }: CategorySelectionModalProps) => {
-  const [searchQuery, setSearchQuery] = useState<string>('')
-  const { categories } = useContext<ICategoriesContext>(CategoriesContext)
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const { categories } = useContext<ICategoriesContext>(CategoriesContext);
+
   const filteredCategories = useMemo(
     () =>
       categories?.filter((c) =>
-        c.name
-          .toLocaleLowerCase()
-          .includes(searchQuery?.toLocaleLowerCase() ?? ''),
+        c.name.toLocaleLowerCase().includes(searchQuery.toLocaleLowerCase() ?? '')
       ),
-    [searchQuery, categories],
-  )
+    [searchQuery, categories]
+  );
 
-  const [isNewCategoryModalOpen, setIsNewCategoryModalOpen] =
-    useState<boolean>(false)
+  const [isNewCategoryModalOpen, setIsNewCategoryModalOpen] = useState<boolean>(false);
 
   const handleOnClick = () => {
-    setIsNewCategoryModalOpen(true)
-  }
+    setIsNewCategoryModalOpen(true);
+  };
 
-  const handleCategorySelection: CategorySelectionFunction = (
-    categoryId,
-    operation,
-  ) => {
+  const handleCategorySelection: CategorySelectionFunction = (categoryId, operation) => {
     switch (operation) {
       case 'add':
-        onCategoriesChange([...categoryIds, categoryId])
-        break
+        onCategoriesChange([...categoryIds, categoryId]);
+        break;
       case 'remove':
-        onCategoriesChange(categoryIds.filter((id) => id !== categoryId))
-        break
+        onCategoriesChange(categoryIds.filter((id) => id !== categoryId));
+        break;
     }
-  }
+  };
 
   return (
     <FullscreenModal>
       <FullscreenModal.Header>
         <ArrowLeft size={24} onClick={onClose} />
-        <FullscreenModal.Title className={classes.modalTitle}>
-          Category Selection
-        </FullscreenModal.Title>
+        <div className={classes.headerContainer}>
+          <FullscreenModal.Title className={classes.modalTitle}>Labels</FullscreenModal.Title>
+          <Button onClick={handleOnClick} className={classes.saveButton}>
+            Save
+          </Button>
```

Comment:
The save button should be removed, since the changes you make are already saved to the activity